lib/jaeger/client/tracer.rb

```diff
@@ -148,9 +144,9 @@ def inject(span_context, format, carrier)
       def extract(format, carrier)
         case format
         when OpenTracing::FORMAT_TEXT_MAP
-          parse_context(carrier['uber-trace-id'])
+          @propagation_codec.extract_text_map(carrier)
         when OpenTracing::FORMAT_RACK
-          parse_context(carrier['HTTP_UBER_TRACE_ID'])
+          @propagation_codec.extract_rack(carrier)
```

Comment:
I think we can forward all this completely. Basically:
```
def extract(format, carrier)
  @propagation_codec.extract(format, carrier)
end
```
This is similar to my first comment in the tracer initialization code. By having PropagationCodec with a simple interface: two methods inject and extract like in the OT API, users can more easily implement their own codec modules/classes if they need to.

In future we may want to extract the propagation logic into a separate gem and use it in [zipkin](https://github.com/salemove/zipkin-ruby-opentracing) client as well.